code/modules/unit_tests/species_unique_id.dm

```diff
@@ -0,0 +1,14 @@
+/**
+ * Every species should use a species ID unique to it and it alone. This test runs through every subtype of /datum/species, and checks for a species ID.
+ * Every ID is written to a list, gathered_species_ids, and if a previously written ID is written again, this test will fail.
+ */
+/datum/unit_test/species_unique_id
+
+/datum/unit_test/species_unique_id/Run()
+	var/list/gathered_species_ids = list()
+	for(var/datum/species/species as anything in subtypesof(/datum/species))
+		var/species_id = initial(species.id)
+		if(gathered_species_ids.Find(species_id))
+			Fail("Duplicate species ID! [species_id] is not unique to a single species.")
+		else
+			gathered_species_ids += species_id
```

Comment:
I think it would be more efficient if you did it with assocs. Something like
```suggestion
	for(var/datum/species/species as anything in subtypesof(/datum/species))
		var/species_id = initial(species.id)
		if(gathered_species_ids[species_id])
			Fail("Duplicate species ID! [species_id] is not unique to a single species.")
		else
			gathered_species_ids[species_id] = TRUE
```
Associative lookups should be something like O(1), while Find should be O(n). If you ended up doing this please be sure to test it again, suggestions are webedits. Nice pr otherwise.